src/commands/duel.ts

```diff
@@ -0,0 +1,287 @@
+          await this.updateUserScore(challengerName, 'win')
+          await this.updateUserScore(accepterName, 'loss')
+
+          winnerText = `${challengerName} has won!`
+        } else if (accepterScore > challengerScore) {
+          await this.updateUserScore(challengerName, 'loss')
+          await this.updateUserScore(accepterName, 'win')
+
+          winnerText = `${accepterName} has won!`
+        } else {
+          await this.updateUserScore(challengerName, 'draw')
+          await this.updateUserScore(accepterName, 'draw')
+
+          winnerText = "It's a draw!"
+        }
+
+        await collectionInteraction.followUp({
+          content: `${accepterName} has rolled a ${accepterScore} and ${challengerName} has rolled a ${challengerScore}. ${winnerText}`,
+        })
+      }
+    })
+  }
+
+  @Slash('duelstats')
+  private async duelStats(interaction: CommandInteraction) {
+    await interaction.deferReply()
+
+    const userName = interaction.user.username
+    const stats = await this.client.duels.findUnique({
+      where: {
+        userName: userName,
+      },
+    })
+    if (stats) {
+      let statsMessage = []
+      if (stats.winStreak > 0) {
+        // User is currently on a win streak
+        statsMessage = [
+          `${userName}: ${stats.wins}-${stats.losses}-${stats.draws}`,
+          `Win streak: ${stats.winStreak}`,
+          `Best win streak: ${stats.winStreakMax}`,
+          `Worst loss streak: ${stats.lossStreakMax}`,
+        ]
+      } else if (stats.lossStreak > 0) {
+        // User is currently on a loss streak
+        statsMessage = [
+          `${userName}: ${stats.wins}-${stats.losses}-${stats.draws}`,
+          `Loss streak: ${stats.lossStreak}`,
+          `Best win streak: ${stats.winStreakMax}`,
+          `Worst loss streak: ${stats.lossStreakMax}`,
+        ]
+      } else {
+        // User's last duel was a draw which reset both streaks
+        statsMessage = [
+          `${userName}: ${stats.wins}-${stats.losses}-${stats.draws}`,
+          `Your last duel was a draw.`,
+          `Best win streak: ${stats.winStreakMax}`,
+          `Worst loss streak: ${stats.lossStreakMax}`,
+        ]
+      }
+      await interaction.followUp(statsMessage.join('\n'))
+    } else {
+      await interaction.followUp(`${userName}, you have never duelled before.`)
+    }
+  }
+
+  private async updateUserScore(userName: string, outcome: 'win' | 'loss' | 'draw') {
+    const user = await this.client.duels.findUnique({
+      where: { userName: userName },
+    })
+    switch (outcome) {
+      case 'draw': {
+        await this.client.duels.upsert({
+          where: {
+            userName: userName,
+          },
+          update: {
+            draws: { increment: 1 },
+            lossStreak: 0,
+            winStreak: 0,
+          },
+          create: {
+            userName: userName,
+            draws: 1,
+          },
+        })
+        break
+      }
+      case 'win': {
```

Comment:
Should probably change `win` or `loss` to use upsert to consistent. (or vice versa)

Reply:
I'm now passing the duel stats each time which means I don't have to create anything here.